lib/array_intersection.rb

```diff
@@ -1,6 +1,23 @@
 # Returns a new array to that contains elements in the intersection of the two input arrays
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the length of array1
+# Space complexity: O(n) where n is the length of the intersection array
 def intersection(array1, array2)
-  raise NotImplementedError
+  intersection = []
+  return intersection if array1 == nil ||
+                         array2 == nil ||
+                         array1 == [] ||
+                         array2 == []
+
+  i = 0
+  while array1[i] != nil
+    j = 0
+    while array2[j] != nil
+      if array1[i] == array2[j]
+        intersection << array2[j]
```

Comment:
If you're not looking for duplicate entries, you should be able to break out of the inner loop at this point.